examples/sphere/opt_limiters_solidbody.jl

```diff
@@ -0,0 +1,317 @@
+
+"""
+    convergence_rate(err, Δh)
+
+Estimate convergence rate given vectors `err` and `Δh`
+
+    err = C Δh^p+ H.O.T
+    err_k ≈ C Δh_k^p
+    err_k/err_m ≈ Δh_k^p/Δh_m^p
+    log(err_k/err_m) ≈ log((Δh_k/Δh_m)^p)
+    log(err_k/err_m) ≈ p*log(Δh_k/Δh_m)
+    log(err_k/err_m)/log(Δh_k/Δh_m) ≈ p
+
+"""
+convergence_rate(err, Δh) =
+    [log(err[i] / err[i - 1]) / log(Δh[i] / Δh[i - 1]) for i in 2:length(Δh)]
+
+# Advection problem on a sphere with bounds-preserving quasimonotone limiter.
+# The initial condition can be set via a command line argument.
+# Possible test cases are: cosine_bells (default), gaussian_bells, and cylinders
+
+const R = 6.37122e6  # sphere radius
+const r0 = R / 2     # bells radius
+const D₄ = 6.6e14    # hyperdiffusion coefficient
+const u0 = 2 * pi * R / (86400 * 12)
+const T = 86400 * 12 # simulation period in seconds (12 days)
+const centers = [
+    Geometry.LatLongPoint(0.0, rad2deg(5 * pi / 6) - 180.0),
+    Geometry.LatLongPoint(0.0, rad2deg(7 * pi / 6) - 180.0),
+] # center of bells
+const test_name = get(ARGS, 1, "cosine_bells") # default test case to run
+const cosine_test_name = "cosine_bells"
+const gaussian_test_name = "gaussian_bells"
+const cylinder_test_name = "cylinders"
+
+const lim_flag = true
+const limiter_tol = 5e-14
+
+# Plot variables and auxiliary function
+ENV["GKSwstype"] = "nul"
+import ClimaCorePlots, Plots
+Plots.GRBackend()
+dirname = "cg_sphere_solidbody_limiter_$(test_name)"
+if lim_flag == false
+    dirname = "$(dirname)_no_lim"
+end
+path = joinpath(@__DIR__, "output", dirname)
+mkpath(path)
+
+function linkfig(figpath, alt = "")
+    # buildkite-agent upload figpath
+    # link figure in logs if we are running on CI
+    if get(ENV, "BUILDKITE", "") == "true"
+        artifact_url = "artifact://$figpath"
+        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+    end
+end
+
+FT = Float64
+ne_seq = (10, 20, 40)
+Δh = zeros(FT, length(ne_seq))
+L1err, L2err, Linferr = zeros(FT, length(ne_seq)),
+zeros(FT, length(ne_seq)),
+zeros(FT, length(ne_seq))
+Nq = 4
+
+# h-refinement study
+for (k, ne) in enumerate(ne_seq)
+    # Set up space
+    domain = Domains.SphereDomain(R)
+    mesh = Meshes.EquiangularCubedSphere(domain, ne)
+    grid_topology = Topologies.Topology2D(mesh)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+
+    # Initialize variables needed for limiters
+    n_elems = Topologies.nlocalelems(space.topology)
+    min_Q = zeros(n_elems)
+    max_Q = zeros(n_elems)
+
+    coords = Fields.coordinate_field(space)
+    Δh[k] = 2 * R / ne
+    global_geom = space.global_geometry
+
+    y0 = map(coords) do coord
+
+        ϕ = coord.lat
+        λ = coord.long
+        rd = Vector{Float64}(undef, 2)
+
+        for i in 1:2
+            rd[i] = Geometry.great_circle_distance(coord, centers[i], global_geom)
+        end
+
+        # Initialize tracer mixing ratio
```

Comment:
```suggestion
        # Initialize specific tracer concentration
```
Note: mixing ratio usually refers to the mass of tracer per mass of dry air. Since your examples are dry, this happens to be what you have. But more generally, we'd want the specific tracer concentration, which is the tracer mass per mass of moist air.